server/sonar-web/src/main/js/components/SourceViewer/components/CoveragePopup.tsx

```diff
@@ -0,0 +1,153 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import * as React from 'react';
+import { groupBy } from 'lodash';
+import { getTests } from '../../../api/components';
+import { SourceLine, TestCase } from '../../../app/types';
+import BubblePopup from '../../common/BubblePopup';
+import TestStatusIcon from '../../shared/TestStatusIcon';
+import { translate } from '../../../helpers/l10n';
+import { collapsePath } from '../../../helpers/path';
+
+interface Props {
+  branch: string | undefined;
+  componentKey: string;
+  line: SourceLine;
+  onClose: () => void;
+  popupPosition?: any;
+}
+
+interface State {
+  loading: boolean;
+  testCases: TestCase[];
+}
+
+export default class CoveragePopup extends React.PureComponent<Props, State> {
+  mounted = false;
+  state: State = { loading: true, testCases: [] };
+
+  componentDidMount() {
+    this.mounted = true;
+    this.fetchTests();
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    // TODO use branchLike
+    if (
+      prevProps.branch !== this.props.branch ||
+      prevProps.componentKey !== this.props.componentKey ||
+      prevProps.line.line !== this.props.line.line
+    ) {
+      this.fetchTests();
+    }
+  }
+
+  componentWillUnmount() {
+    this.mounted = false;
+  }
+
+  fetchTests = () => {
+    this.setState({ loading: true });
+    getTests(this.props.componentKey, this.props.line.line, this.props.branch).then(
+      testCases => {
+        if (this.mounted) {
+          this.setState({ loading: false, testCases });
+        }
+      },
+      () => {
+        if (this.mounted) {
+          this.setState({ loading: false });
+        }
+      }
+    );
+  };
+
+  handleTestClick = (event: React.MouseEvent<HTMLAnchorElement>) => {
+    event.preventDefault();
+    event.currentTarget.blur();
+    const { key } = event.currentTarget.dataset;
+    const Workspace = require('../../workspace/main').default;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: require() style import is forbidden ([typescript:S3533](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS3533))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fstas%2Fpopups&issues=AWHb6qaww6XGHKvPuQTS&open=AWHb6qaww6XGHKvPuQTS)

[ISSUE_KEY]: AWHb6qaww6XGHKvPuQTS